### PR TITLE
just a CI test

### PR DESCRIPTION
**Pull request by m-schmoock**
Just a test PR, trying to debug some CI issues...

### CODE REVIEW

**Comment by cdecker** [comment]
Are you sure Github just renamed the branch without prompting us? That'd be a very surprising move and definitely cause a lot of trouble. I just checked out this repo and the main branch is still called `master`.

**Comment by m-schmoock** [comment]
@cdecker Yes, the make script CI is calling is definitively not working correctly (anymore?)
Current master source results in the following CI output when doing `make check-source`

```
origin	https://github.com/ElementsProject/lightning (fetch)
origin	https://github.com/ElementsProject/lightning (push)
* (HEAD detached at pull/5913/merge)
fatal: ambiguous argument 'main': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

Currently fiddling around to fix that.
What I simply do is a `git fetch origin/master` and do the diffs against that. This way we don't need the if/else main/master foo.
I'll keep you updated once I have something working...

**Comment by cdecker** [comment]
That usually means that there is both a branch `main` and a file/directory `main` and `git` can't uniquely identify which you meant. Let me check that :-)

**Comment by cdecker** [comment]
Ok, that theory falls through quickly: no `main` branch nor a `main` file.

File: doc/Makefile [hunk 1]
```diff
@@ -209,11 +209,11 @@ doc/index.rst: $(MANPAGES:=.md)
 
 # So GitHub renamed master to main.  This is painful.
 schema-added-check:
-	@if ! git describe master >/dev/null 2>&1; then MASTER=main; else MASTER=master; fi; if git diff $$MASTER doc/schemas | grep -q '^+.*{' && ! git diff master doc/schemas | grep -q '^+.*"added"'; then echo 'New schema fields must have "added": "vNEXTVERSION"' >&2; exit 1; fi
+	@git fetch origin master ; if git diff origin/master...HEAD -- doc/schemas | grep -q '^+.*{' && ! git diff origin/master...HEAD -- doc/schemas | grep -q '^+.*"added"'; then git diff origin/master...HEAD -- doc/schemas ; echo 'New schema fields must have "added": "vNEXTVERSION"' >&2; exit 1; fi
```
Comment (cdecker) [inline doc/Makefile:212]: `git fetch` in a script will cause people headaches if they have a passphrase or a hardware key holding the SSH keys.

Comment (m-schmoock) [reply doc/Makefile:212]: Yep, can't do it that way... :/

**Comment by m-schmoock** [comment]
@cdecker looks like GitHub CI is not checking out the repo with any refs except the PRs branch itself. I think the master/main naming issue doesn't apply for us, as our repo has 'master' branch and it will likely stay that way. The political correct 'main' branch is only for new projects. So we can remove that main/master if/else from our script.

I will try to change it in a way that only when it detects that it is running on GitHub actions CI it will do the 'fetch origin master'.

**Pull request closed without merging**